identity/src/main/java/com/android/identity/CborUtil.java

```diff
@@ -0,0 +1,305 @@
+package com.android.identity;
+
+import static java.lang.annotation.RetentionPolicy.SOURCE;
+
+import androidx.annotation.IntDef;
+import androidx.annotation.NonNull;
+
+import java.io.ByteArrayInputStream;
+import java.lang.annotation.Retention;
+import java.text.DecimalFormat;
+import java.text.DecimalFormatSymbols;
+import java.util.Collection;
+import java.util.List;
+import java.util.Locale;
+
+import co.nstant.in.cbor.CborDecoder;
+import co.nstant.in.cbor.CborException;
+import co.nstant.in.cbor.model.AbstractFloat;
+import co.nstant.in.cbor.model.ByteString;
+import co.nstant.in.cbor.model.DataItem;
+import co.nstant.in.cbor.model.DoublePrecisionFloat;
+import co.nstant.in.cbor.model.NegativeInteger;
+import co.nstant.in.cbor.model.SimpleValue;
+import co.nstant.in.cbor.model.UnsignedInteger;
+
+class CborUtil {
+    private static final String TAG = "CborUtil";
+
+
+    static final int DIAGNOSTICS_FLAG_EMBEDDED_CBOR = (1 << 0);
+
+    static final int DIAGNOSTICS_FLAG_PRETTY_PRINT = (1 << 1);
+
+    @Retention(SOURCE)
+    @IntDef(
+            flag = true,
+            value = {
+                    DIAGNOSTICS_FLAG_EMBEDDED_CBOR,
+                    DIAGNOSTICS_FLAG_PRETTY_PRINT
+            })
+    public @interface DiagnosticsFlags {
+    }
+
+    // Returns true iff all elements in |items| are not compound (e.g. an array or a map).
+    private static boolean cborAreAllDataItemsNonCompound(@NonNull List<DataItem> items,
+                                                          @DiagnosticsFlags int flags) {
+        for (DataItem item : items) {
+            if ((flags & CborUtil.DIAGNOSTICS_FLAG_EMBEDDED_CBOR) != 0 && item.hasTag()) {
```

Comment:
Shouldn't this apply only to tagged bstrs, not any major type with a tag?

Reply:
Nice catch, fixed.